ui/src/app/tasks/task-apps/task-apps.component.spec.ts

```diff
@@ -0,0 +1,65 @@
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+import { DebugElement } from '@angular/core';
+import { By } from '@angular/platform-browser';
+import { BusyModule } from 'tixif-ngx-busy';
+import { NgxPaginationModule } from 'ngx-pagination';
+import { ToastyService } from 'ng2-toasty';
+import { Router } from '@angular/router';
+import { TasksService } from '../tasks.service';
+import { AppRegistration } from '../../shared/model/app-registration';
+import { ApplicationType } from '../../shared/model/application-type';
+import { TaskAppsComponent } from './task-apps.component';
+import { MockToastyService } from '../../tests/mocks/toasty';
+import { MockTasksService } from '../../tests/mocks/tasks';
+
+describe('TaskAppsComponent', () => {
+  let component: TaskAppsComponent;
+  let fixture: ComponentFixture<TaskAppsComponent>;
+  let de: DebugElement;
+  let el: HTMLElement;
+  const toastyService = new MockToastyService();
+  const tasksService = new MockTasksService();
+
+  beforeEach(async(() => {
+    const routerStub = {};
+    TestBed.configureTestingModule({
+      declarations: [
+        TaskAppsComponent
+      ],
+      imports: [
+        BusyModule,
+        NgxPaginationModule
+      ],
+      providers: [
+        { provide: TasksService, useValue: tasksService },
+        { provide: ToastyService, useValue: toastyService },
+        { provide: Router, useValue: routerStub }
+      ]
+    })
+      .compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(TaskAppsComponent);
+    component = fixture.componentInstance;
+    tasksService.testTaskAppRegistrations = [
+      new AppRegistration('fakename', ApplicationType.task, 'fakeuri')
+    ];
+    // tasksService = TestBed.get(TasksService);
+    toastyService.clearAll();
+    fixture.detectChanges();
```

Comment:
Wonder if we want to move the detectChanges into the `it`.  Because future `it` s may have additional configuration before the detect changes.